web/src/hotspots.js

```diff
@@ -273,6 +273,27 @@ function close_read_hotspots(new_hotspots) {
     }
 }
 
+export function open_popover_if_hotspot_exist(hotspot_name, bind_element = null) {
+    const overlay_name = "hotspot_" + hotspot_name + "_overlay";
+
+    if (is_hotspot_exist(hotspot_name)) {
```

Comment:
I don't understand why we need this check -- isn't this function called from a click handler that runs when someone clicks on the hotspot element in the DOM? It seems like it'll always be true by construction, and I'm not sure I see a case where we'd call this function from another code path and want it to not open.

Reply:
According to my understanding, we will also call this function for our non-intro hotspots, which will have custom triggers. So sometimes it could be the case that a developer may try to open a hotspot that's already has been seen and hence not exist.